nvme.c

```diff
@@ -8357,8 +8242,6 @@ static int passthru(int argc, char **argv, bool admin,
 	}
 free_metadata:
 	free(mdata);
-free_data:
-	nvme_free_huge(data, huge);
 close_dfd:
```

Comment:
Sorry this is not related to the changes but does not work correctly as the `close_dfd:` is not executed after the `close_mfd:` but `goto close mfd;` is executed after the `dfd` opened so needed to execute the `close_dfd:` also.

```c
	if (strlen(cfg.input_file)) {
		dfd = open(cfg.input_file, flags, mode);
		if (dfd < 0) {
			nvme_show_perror(cfg.input_file);
			return -EINVAL;
		}
	}

	if (cfg.metadata && strlen(cfg.metadata)) {
		mfd = open(cfg.metadata, flags, mode);
		if (mfd < 0) {
			nvme_show_perror(cfg.metadata);
			err = -EINVAL;
			goto close_dfd;
		}
	}

	if (cfg.metadata_len) {
		mdata = malloc(cfg.metadata_len);
		if (!mdata) {
			err = -ENOMEM;
			goto close_mfd;
		}
...
free_metadata:
	free(mdata);
close_dfd:
	if (strlen(cfg.input_file))
		close(dfd);
close_mfd:
	if (cfg.metadata && strlen(cfg.metadata))
		close(mfd);
```
By the way I think only one goto label is needed for the function as combined the three goto labels as below for example. Anyway probably this may be needed to be fixed by other PR changes.
```c
	void *mdata = NULL;
	int dfd = -1;
	int mfd = -1;
...
	if (!***) {
		err = -E***;
		goto ret;
	}
...
ret:
	free(mdata);
	if (dfd != -1 && dfd != STDIN_FILENO && dfd != STDOUT_FILENO)
		close(dfd);
	if (mfd != -1 && mfd != STDIN_FILENO && mfd != STDOUT_FILENO)
		close(mfd);
...

Reply:
Good catch! I missed the `passthru` function to use the cleanup helpers as well.

Reply:
Fixed in the new version.